### PR TITLE
Add parser for 'candlepin/error.log'

### PR DESCRIPTION
**Pull request by sagaraivale**

### CODE REVIEW

**Comment by sagaraivale** [comment]
@vishwanathjadhav @xiangce  
Kindly review parser for 'sos_commands/foreman/foreman-debug/var/log/candlepin/error.log'

File: insights/config/specs.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
                                     SimpleFileSpec("sos_commands/filesys/blkid")]),
     "bond"                      : PatternSpec(r"proc/net/bonding/bond.*"),
     "brctl_show"                : CommandSpec("/usr/sbin/brctl show"),
+    "candlepin_error_log"       : SimpleFileSpec(r"sos_commands/foreman/foreman-debug/var/log/candlepin/error.log", large_content=True),
```
Comment (bfahr) [inline insights/config/specs.py]: @sagaraivale would you mind to move your new spec after `candlepin.log`?

File: insights/parsers/foreman_log.py [hunk 2]
```diff
@@ -38,3 +38,39 @@ class ProductionLog(LogFileOutput):
 class CandlepinLog(LogFileOutput):
     """Class for parsing ``candlepin/candlepin.log`` file."""
     pass
+
+
+@parser('candlepin_error_log')
+class Candlepin_Error_Log(LogFileOutput):
```
Comment (bfahr) [inline insights/parsers/foreman_log.py]: Class names should be camel-case with no underscores.

File: insights/tests/test_foreman_log.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
+from insights.tests import context_wrap
```
Comment (bfahr) [inline insights/tests/test_foreman_log.py]: It appears you duplicated this file from `insights/parsers/tests/test_foreman_log.py`.  You should add your tests to the existing file and not add this one.

File: insights/parsers/foreman_log.py [hunk 4]
```diff
@@ -38,3 +38,39 @@ class ProductionLog(LogFileOutput):
...
+        org.candlepin.sync.ImportConflit Exception: Owner has already imported from another subscription management application.']
+
+        >>> candlepin_log.get_after(datetime(2016, 9, 7, 16, 0, 0)
+        ['2016-09-07 16:49:24,650 [=, org=] WARN  org.apache.qpid.transport.network.security.ssl.SSLUtil - Exception received while trying to verify hostname',
+        '2016-09-07 18:7:53,688 [req=d9dc3cfd-abf7-485e-b1eb-e1e28e4b0f28, org=org_ray] ERROR org.candlepin.sync.Importer - Conflicts occurred during import that
+        were not overridden:', '2016-09-07 18:07:53,690 [req=d9dc3cfd-abf7-485e-b1eb-e1e28e4b0f28, org=org_ray] ERROR org.candlepin.sync.Importer - [DISTRIBUTOR_CONFLICT]',
+        '2016-09-07 18:07:53,711 [req=d9dc3cfd-abf7-485e-b1eb-e1e28e4b0f28, org=org_ray] ERROR org.candlepin.resource.OwnerResource - Recording import failure
+        org.candlepin.sync.ImportConflictException: Owner has already imported from another subscription management application.']
+
+  """
```
Comment (bfahr) [inline insights/parsers/foreman_log.py]: Please add a pass statement to the end of this class.

File: insights/config/specs.py [hunk 5]
```diff
@@ -20,9 +20,9 @@
                                     SimpleFileSpec("sos_commands/filesys/blkid")]),
     "bond"                      : PatternSpec(r"proc/net/bonding/bond.*"),
     "brctl_show"                : CommandSpec("/usr/sbin/brctl show"),
-    "candlepin_error_log"       : SimpleFileSpec(r"sos_commands/foreman/foreman-debug/var/log/candlepin/error.log", large_content=True),
     "candlepin.log"             : First([SimpleFileSpec(r"var/log/candlepin/candlepin.log", large_content=True),
                                     SimpleFileSpec(r"sos_commands/foreman/foreman-debug/var/log/candlepin/candlepin.log", large_content=True)]),
+    "candlepin_error_log"       : SimpleFileSpec(r"sos_commands/foreman/foreman-debug/var/log/candlepin/error.log", large_content=True),
```
Comment (xiangce) [inline insights/config/specs.py]: Better user NoneGroup to wrap this SimpleFileSpec, which to indicate the client tool does not try to collect this file.

File: insights/tests/test_foreman_log.py (hunk 3 above)
Comment (bfahr) [reply insights/tests/test_foreman_log.py]: @sagaraivale you should remove this file and add your new tests to the file `insights/parsers/tests/test_foreman_log.py`.  All parser tests should be in the `tests` folder under `parsers`.

Comment (bfahr) [reply insights/tests/test_foreman_log.py]: Please remove this file.

File: insights/parsers/tests/test_candlepin_error_log.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+from insights.tests import context_wrap
```
Comment (bfahr) [inline insights/parsers/tests/test_candlepin_error_log.py]: Normally this test would be added to the module test file `parsers/tests/test_foreman_log.py`, is there a reason you created a separate test file?

Comment (sagaraivale) [reply insights/parsers/tests/test_candlepin_error_log.py]: @bfahr  Write the test case under ```parsers/tests/test_foreman_log.py``` file  and delete extra ```parsers/tests/test_candlepin_error_log.py``` file. 
Kindly review...!

File: insights/tests/test_foreman_log.py (hunk 3 above)
Comment (sagaraivale) [reply insights/tests/test_foreman_log.py]: @bfahr  I removed ```insights/parsers/tests/test_foreman_log.py```. Kindly review...!

**Pull request merged**